src/odemis/dataio/tiff.py

```diff
@@ -2722,3 +2750,188 @@ def _convert_thermo_fisher_to_odemis_metadata(metadata: str) -> dict:
         logging.warning(f"Failed to parse ThermoFisher metadata: {model.MD_DET_TYPE} - {e}")
 
     return md
+
+def _parse_zeiss_metadata(metadata:str) -> dict:
+    """
+    Parses ZEISS metadata to dictionary
+    :param metadata: metadata string
+    :return: (dict) metadata as dictionary
+    """
+    zmd = {}
+    for line in metadata.split("\r\n"):
+        # Metadata is in the format:
+        # HEADER
+        # key = value unit
+
+        # Example:
+        # AP_STAGE_AT_X
+        # Stage at X = 48.5090 mm
+
+        try:
+            
+            # skip all caps lines as they are headers
+            if line.isupper():
+                continue
+
+            # NOTE: 
+            # Date, Time are separated by : 
+            # everything else is separated by =
+            try:
+                k, v = line.split("=")
+            except ValueError:             
+                k, v = line.split(":", maxsplit=1) 
+            k, v = k.strip(), v.strip()
+            zmd[k] = v
+        except ValueError:
+            continue  # skip lines that don't contain key value pairs
+        except Exception as e:
+            # any other error -> something has gone wrong
+            logging.warning(f"Failed to parse ZEISS metadata: {line} - {e}", exc_info=True)
+
+    return zmd
+
+
+def _convert_zeiss_to_odemis_metadata(metadata: str) -> dict:
+    """Converts ZEISS metadata to odemis metadata format
+    :param metadata: metadata string
+    :return: (dict) metadata as dictionary
+    """
+
+    # parse as dict
+    # from pprint import pprint
+    zmd = _parse_zeiss_metadata(metadata)
+
+    md = {}
+
+    # general
+    try:
+        datetime_str = zmd["Date"] + " " + zmd["Time"]
+        date = datetime.strptime(datetime_str, "%d %b %Y %H:%M:%S")
+        md[model.MD_ACQ_DATE] = date.timestamp()
+    except Exception as e:
+        logging.warning(f"Failed to parse ZEISS metadata: {model.MD_ACQ_DATE} - {e}")
+    try:
+        md[model.MD_HW_NAME] = zmd["Serial No."]
+    except Exception as e:
+        logging.warning(f"Failed to parse ZEISS metadata: {model.MD_HW_NAME} -  {e}")
+    try:
+        md[model.MD_SW_VERSION] = zmd["Version"]
+    except Exception as e:
```

Comment:
Why do you use generic exceptions here and not for instance a `KeyError`?

Reply:
I've change the above two to KeyError but think the rest should stay as generic exceptions because there are multiple ways the parser can fail, e.g. key error, formatting, additional chars.

Reply:
I've changed the two above to KeyError, but think the rest should stay as generic exceptions because there are multiple things that can fail for those keys.